src/transformers/models/dbrx/modeling_dbrx.py

```diff
@@ -495,9 +499,33 @@ def _flash_attention_forward(
 
             attn_output = pad_input(attn_output_unpad, indices_q, batch_size, query_length)
         else:
-            attn_output = flash_attn_func(
-                query_states, key_states, value_states, dropout, softmax_scale=softmax_scale, causal=causal
-            )
+            if (position_ids[:,-1]==position_ids.size(1)-1).all():
```

Comment:
this is an input dependent control flow which won't be supported by compile, but compile is already not supported. 
This needs a comment: we are checking that the input is not padded, and that we are doing prefill right?